src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -67,28 +94,60 @@ public EditCommand(Index index, EditPersonDescriptor editPersonDescriptor) {
 
         this.index = index;
         this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
+        this.editTaskDescriptor = null;
+        this.isTask = false;
+    }
+
+    /**
+     * @param index of the task in the filtered task list to edit
+     * @param editTaskDescriptor details to edit the task with
+     */
+    public EditCommand(Index index, EditTaskDescriptor editTaskDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editTaskDescriptor);
+
+        this.index = index;
+        this.editTaskDescriptor = new EditTaskDescriptor(editTaskDescriptor);
+        this.editPersonDescriptor = null;
+        this.isTask = true;
+        //TODO see if we can use Singleton pattern to handle the Descriptors.
```

Comment:
An alternative is to have `EditPersonCommand` and `EditTaskCommand` to inherit from `EditCommand`

Reply:
This would be a good way to keep the Task vs Person objects apart as well. Thank you for the suggestion!